### PR TITLE
Automate L10n extraction locally and in CI

### PR DESCRIPTION
**Pull request by KevinMind**
Fixes: https://github.com/mozilla/addons/issues/14848

### Testing

PR on origin: [Link](https://github.com/mozilla/addons-frontend/actions/runs/9905346462/job/27364700229?pr=12800)
PR on fork: TBD

### CODE REVIEW

**Comment by KevinMind** [comment]
Generated this PR: https://github.com/mozilla/addons-frontend/pull/12801

**Comment by KevinMind** [comment]
I've merged all of the scripts into a single zx executable. It extracts and merges locales for the current code base. It is pretty fast 2x the previous set of scripts (<15seconds locally and in CI)

Local Options

1. Run the locale.mjs script as a pre-commit hook
2. Run the babel plugin directly in our dev and prod builds

CI options

1. Run on every PR, failing if locales are not built
2. Run on pull requests, pre merge (may need queue to do that I'm not sure)
3. Run on master commits, post merge but pre tag
  a. we can commit directly on master
  b. we could open a PR with the generated locale changes.

All options would work, it's more about the kind of experience we want to have.

File: docs/i18n.md [hunk 1]
```diff
@@ -20,96 +20,11 @@ NODE_PATH='./:./src' bin/create-locales
...
-
-Now that you have extracted and merged locales for one application, it's time to create a pull request for your branch. For example:
-
-```
-git push origin amo-locales
-```
-
-If the pull request passes all of our CI tests it is likely good to merge. You don't need to ask for a review unless you're unsure of something because often locale updates will be thousands of lines of minor diffs that can't be reasonably reviewed by a human. 🙂 If the pull request passes all of our CI tests it is likely good to merge.
-
-#### Building the JS locale files
```
Comment (KevinMind) [inline docs/i18n.md:103]: It's not totally clear to me what this is for and how it get's run in our local development flow and or in CI. Do I need to integrate this logic to build the json files as a part of `extract-locales` should be simple enough but I'm not sure if this is needed.

File: babel.config.locales.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          'Project-Id-Version': 'amo',
+          'Report-Msgid-Bugs-To': 'EMAIL@ADDRESS',
+          'POT-Creation-Date': potCreationDate,
+          'PO-Revision-Date': 'YEAR-MO-DA HO:MI+ZONE',
+          'Last-Translator': 'FULL NAME <EMAIL@ADDRESS>',
+          'Language-Team': 'LANGUAGE <LL@li.org>',
+          'MIME-Version': '1.0',
+          'Content-Type': 'text/plain; charset=utf-8',
+          'Content-Transfer-Encoding': '8bit',
+          'plural-forms': 'nplurals=2; plural=(n!=1);',
```
Comment (eviljeff) [inline babel.config.locales.js:23]: some of this looks a bit boilerplate-y (e.g. `EMAIL@ADDRESS`) - does it need updating?

**Review by eviljeff — APPROVED** [review]
r+wc

File: babel.config.locales.js (hunk 2 above)
Comment (KevinMind) [reply babel.config.locales.js:23]: Copied verbatim from [here](https://github.com/mozilla/addons-frontend/blob/master/webpack.l10n.config.babel.js#L24-L55). I could actually remove the webpack config file while we are here as that is no longer referenced or needed.

**Review by willdurand — APPROVED** [review]

**Pull request merged**